src/core/thread/dua_manager.cpp

```diff
@@ -476,7 +476,7 @@ void DuaManager::PerformNextRegistration(void)
         const Ip6::Address *duaPtr = nullptr;
         Child *             child  = nullptr;
 
-        if (!mRegisterCurrentChildIndex)
+        if (!mRegisterCurrentChildIndex || !mChildDuaMask.Get(mChildIndexDuaRegistering))
```

Comment:
keep it as is and reset `mRegisterCurrentChildIndex` to false in below ?https://github.com/openthread/openthread/pull/6637/files#diff-f22621e2122bf5e36a030cbcbfbe861983e321e66d3e38921632626b9eb3b47aR664

`mRegisterCurrentChildIndex` is designed to directly reuse the `mChildIndexDuaRegistering`  and skip the unnecessary iteration below if we know the index to register already.

Reply:
`mRegisterCurrentChildIndex` related logic is now removed.

Reply:
Thanks for cleaning up this. 👍